vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -362,6 +367,19 @@ func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isRead
 
 // private function that does the job of mounting volume in conjunction with refcounting
 func (d *VolumeDriver) processMount(r volume.MountRequest) volume.Response {
+	// get volume metadata
+	status, err := d.GetVolume(r.Name)
```

Comment:
you need GetVolume on short names only, so the check for IsFullVolumeName should happen before the GetVolume .

 I think the algorithm is supposed to be as follows (for mount , in pseudocode)
```

  if IsFullVolumeName(name) {
         name = r.Name
  } else {
     s.GetVolume () //get volume info 
     name = FormFullVolumeName (r.Name, GetDatastore(result_of_volume_get)
  }
```